classes/migration/upgrade/v3_5_0/I11673_AddMissingApprovalToReviewerSuggestion.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_5_0/I11673_AddMissingApprovalToReviewerSuggestion.php
+ *
+ * Copyright (c) 2014-2025 Simon Fraser University
+ * Copyright (c) 2000-2025 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I11673_AddMissingApprovalToReviewerSuggestion
+ *
+ * @brief Adds missing approval to reviewer suggestion
+ * @see https://github.com/pkp/pkp-lib/issues/11673
+ */
+
+namespace PKP\migration\upgrade\v3_5_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\migration\Migration;
+use PKP\submission\reviewer\suggestion\ReviewerSuggestion;
+
+class I11673_AddMissingApprovalToReviewerSuggestion extends Migration
+{
+    public function up(): void
+    {
+        // Get all those reviewer suggestion for which there is not approver and no reviewer yet ,
+        // but there is a same email address match user and there is a review assignment for 
+        // that user and the submission as the reviewer suggestion .
+        $suggestions = DB::table('reviewer_suggestions')
+            ->join('users', 'reviewer_suggestions.email', '=', 'users.email')
+            ->join('review_assignments', function ($join) {
+                $join
+                    ->on(
+                        'users.user_id',
+                        '=',
+                        'review_assignments.reviewer_id'
+                    )
+                    ->on(
+                        'reviewer_suggestions.submission_id',
+                        '=',
+                        'review_assignments.submission_id'
+                    );
+            })
+            ->whereNull('reviewer_suggestions.approved_at')
+            ->whereNull('reviewer_suggestions.reviewer_id')
+            ->select('reviewer_suggestions.*', 'users.user_id', 'review_assignments.review_id')
+            ->get();
+
+        $primaryKey = (new ReviewerSuggestion())->getKeyName();
```

Comment:
It would be better not to instantiate the model in the migration scripts; just hard-code `reviewer_suggestion_id` below. This will work for now, but down the road it might cause e.g. model lifecycle events to happen before a migration has run to adjust the schema. As much as possible, migration code should not interact with application code.